File: spring/web/SpringMvcTesting/src/main/java/com/instil/mvc/services/QuotesController.java
package com.instil.mvc.services;

import com.instil.mvc.logic.QuotesEngine;
import com.instil.mvc.model.Quote;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import static org.springframework.http.ResponseEntity.notFound;

@RestController
@RequestMapping("/quotes")
public class QuotesController {
    @Autowired
    private QuotesEngine engine;

    @GetMapping(value="{courseId}", produces = "application/json")
    public ResponseEntity<Quote> fetchQuoteAsJson(@PathVariable String courseId) {
        return engine.generateQuote(courseId)
                .map(ResponseEntity::ok)
                .orElseGet(() -> notFound().build());
    }
}

File: spring/web/SpringMvcTestingKotlin/src/main/kotlin/demos/spring/boot/rest/services/QuotesController.kt
package demos.spring.boot.rest.services

import demos.spring.boot.rest.logic.QuotesEngine
import demos.spring.boot.rest.model.Quote
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.http.ResponseEntity.*
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/quotes")
class QuotesController(@Autowired val engine: QuotesEngine) {

    @GetMapping(value = ["{courseId}"], produces = ["application/json"])
    fun fetchQuoteAsJson(@PathVariable courseId: String): ResponseEntity<Quote> {
        val quote = engine.generateQuote(courseId)
        return if(quote != null) ok(quote) else notFound().build()
    }
}
